crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,437 @@
+
+/// Defines the text direction
+///
+/// For example English is written LTR (left-to-right) while Arabic is written RTL (right-to-left).
+#[derive(
+    Component, Copy, Clone, PartialEq, Eq, Debug, Default, Serialize, Deserialize, Reflect,
+)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum TextDirection {
+    /// Inherit from parent node
+    #[default]
+    Inherit,
+    /// Text is written left to right
+    LeftToRight,
+    /// Text is written right to left
+    RightToLeft,
+}
+
+/// Whether to show or hide overflowing items
+#[derive(
+    Component, Copy, Clone, PartialEq, Eq, Debug, Default, Reflect, Serialize, Deserialize,
+)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum Overflow {
+    /// Show overflowing items
+    #[default]
+    Visible,
+    /// Hide overflowing items
+    Hidden,
+}
+
+/// Flexbox-specific layout components
+pub mod flex {
+
+    use bevy_ecs::query::{Changed, Or, WorldQuery};
+
+    use super::*;
+
+    /// A query for all of the components need for flexbox layout.
+    ///
+    /// See [`FlexLayoutChanged`] when attempting to use this as a query filter.
+    #[derive(WorldQuery)]
+    pub struct FlexLayoutQuery {
+        /// The layout algorithm used
+        pub layout_strategy: &'static LayoutStrategy,
+        /// The position of this UI node
+        pub offset: &'static Offset,
+        /// Whether the node should be absolute or relatively positioned
+        pub position_type: &'static PositionType,
+        /// The constraints on the size of this node
+        pub size_constraints: &'static SizeConstraints,
+        /// The margin, padding and border of the UI node
+        pub spacing: &'static Spacing,
+        /// The flexbox layout parameters
+        pub flex_layout: &'static FlexLayout,
+        /// The direction of the text
+        pub text_direction: &'static TextDirection,
+        /// The behavior in case the node overflows its allocated space
+        pub overflow: &'static Overflow,
+    }
+
+    /// A type alias for when any of the components in a [`FlexLayoutQuery`] have changed.
+    pub type FlexLayoutChanged = Or<(
+        Changed<LayoutStrategy>,
+        Changed<PositionType>,
+        Changed<SizeConstraints>,
+        Changed<Spacing>,
+        Changed<FlexLayout>,
+        Changed<TextDirection>,
+        Changed<Overflow>,
+    )>;
+
+    /// The flexbox-specific layout configuration of a UI node
+    ///
+    /// This follows the web spec closely,
+    /// you can use [guides](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) for additional documentation.
+    #[derive(Component, Serialize, Deserialize, Reflect, Debug, PartialEq, Clone, Copy)]
+    #[reflect_value(PartialEq, Serialize, Deserialize)]
+    pub struct FlexLayout {
+        /// How items are ordered inside a flexbox
+        ///
+        /// Sets the main and cross-axis: if this is a "row" variant, the main axis will be rows.
+        pub flex_direction: FlexDirection,
```

Comment:
```suggestion
        pub direction: FlexDirection,
```
To keep consistency with other properties which doesn't have `flex_` prefix